File: Modules/SMES/MES/SIE.Web.MES/WorkOrders/Commands/GeneraDispatchTaskCommand.js
SIE.defineCommand('SIE.Web.MES.WorkOrders.Commands.GeneraDispatchTaskCommand', {
    meta: { text: "生成任务单", group: "edit", iconCls: "icon-AlignBottom icon-blue" },
    canExecute: function (view) {
        if (view != null && view.getSelection().length > 0 && view.getSelection().all(p => p.getState() != 2 && p.getState() != 3 && p.getIsPause() != 1))
            return true;
        return false;
    },
    execute: function (view) {
        var ids = view.getSelectionIds();
        SIE.Msg.wait('生成任务单中......'.t());
        view.execute({
            data: ids,
            success: function (res) {
                SIE.Msg.showInstantMessage('生成结束'.t());
                view.reloadData();
            }
        });
    }
});

File: Modules/SMES/MES/SIE.Web.MES/WorkOrders/Commands/GeneraDispatchTaskCommand.cs
using NPOI.POIFS.Storage;
using SIE.MES.WorkOrders;
using SIE.Web.Command;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SIE.Web.MES.WorkOrders.Commands
{
    public class GeneraDispatchTaskCommand : ViewCommand<List<double>>
    {
        protected override object Excute(List<double> args, string scope)
        {
            RT.Service.Resolve<WorkOrderController>().GeneraDispatchTask(args);
            return true;
        }
    }
}
